src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -110,30 +113,35 @@ public CommandResult execute(Model model) throws CommandException {
             }
         }
 
-        if (personToEdit.getStatus().toString().equals(Status.POSITIVE)
-                && editedPerson.getStatus().toString().equals(Status.NEGATIVE)) {
-
-            List<Person> filteredByClassCodeList = studentList.stream()
-                    .filter(student -> student.getClassCode().toString().equals(editedPerson.getClassCode().toString())
-                            && !student.isSamePerson(editedPerson))
-                    .collect(Collectors.toList());
-
-            List<Person> filteredByPositiveStatusInClass = filteredByClassCodeList.stream()
-                    .filter(student -> student.getStatus().toString().equals(Status.POSITIVE))
-                    .collect(Collectors.toList());
-
-            if (filteredByPositiveStatusInClass.size() == 0) {
-                for (int i = 0; i < filteredByClassCodeList.size(); i++) {
-                    Person currentPerson = filteredByClassCodeList.get(i);
-                    EditPersonDescriptor tempDescriptor = new EditPersonDescriptor();
-                    tempDescriptor.setStatus(new Status(Status.NEGATIVE));
-                    Person editedPersonStatus = createEditedPerson(currentPerson, tempDescriptor);
-                    model.setPerson(currentPerson, editedPersonStatus);
+        try {
+            if (personToEdit.getStatus().toString().equals(Status.POSITIVE)
+                    && editedPerson.getStatus().toString().equals(Status.NEGATIVE)) {
+
+                List<Person> filteredByClassCodeList = studentList.stream()
+                        .filter(student -> student.getClassCode().toString()
+                                .equals(editedPerson.getClassCode().toString())
+                                && !student.isSamePerson(editedPerson))
+                        .collect(Collectors.toList());
+
+                List<Person> filteredByPositiveStatusInClass = filteredByClassCodeList.stream()
+                        .filter(student -> student.getStatus().toString().equals(Status.POSITIVE))
+                        .collect(Collectors.toList());
+
+                if (filteredByPositiveStatusInClass.size() == 0) {
+                    for (int i = 0; i < filteredByClassCodeList.size(); i++) {
+                        Person currentPerson = filteredByClassCodeList.get(i);
+                        EditPersonDescriptor tempDescriptor = new EditPersonDescriptor();
+                        tempDescriptor.setStatus(new Status(Status.NEGATIVE));
+                        Person editedPersonStatus = createEditedPerson(currentPerson, tempDescriptor);
+                        model.setPerson(currentPerson, editedPersonStatus);
+                    }
```

Comment:
Code looks clear and concise. I think it may just be that string method calls can get very long. We might have to consider abstracting this out if we are expanding on this in the future. Otherwise all's good!